app/services/foreman/plugin.rb

```diff
@@ -351,6 +351,12 @@ def parameter_filters(klass)
       @parameter_filters.fetch(klass.is_a?(Class) ? klass.name : klass, [])
     end
 
+    def hosts_controller_action_scope(action, &block)
+      # Add scope for both API and UI controller.
+      HostsController.add_scope_for(action, block)
```

Comment:
It sounds reasonable to me that the info that is returned is similar between the API and UI. In the original bug in Katello I saw that the problem exists in both UI and API.
I won't fight over it though. If you think it's unlikely, I can separate it to different methods.